packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,159 @@
+import type { HTMLAttributes } from 'react';
+import React, { cloneElement, forwardRef, useMemo, useState } from 'react';
+import cn from 'classnames';
+import {
+  useFloating,
+  autoUpdate,
+  offset,
+  flip,
+  shift,
+  arrow,
+  useHover,
+  useFocus,
+  useDismiss,
+  useRole,
+  useInteractions,
+  useTransitionStyles,
+  FloatingArrow,
+  FloatingPortal,
+} from '@floating-ui/react';
+
+import styles from './Tooltip.module.css';
+import mergeRefs from './utils';
+
+const ARROW_HEIGHT = 7;
+const GAP = 4;
+
+export type TooltipProps = {
+  /**
+   * The element that triggers the tooltip.
+   * @note Needs to be a single ReactElement and not: <React.Fragment/> | <></>
+   */
+  children: React.ReactElement & React.RefAttributes<HTMLElement>;
+  content: string;
+  /**
+   * Placement of the tooltip on the trigger.
+   * @default 'top'
+   */
+  placement?: 'top' | 'right' | 'bottom' | 'left';
+  /** Delay in milliseconds before opening.
+   * @default 150
+   */
+  delay?: number;
+  open?: boolean;
+  defaultOpen?: boolean;
+} & HTMLAttributes<HTMLDivElement>;
+
+export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
+  (
+    {
+      children,
+      content,
+      placement = 'top',
+      delay = 150,
+      open: userOpen,
+      defaultOpen = false,
+      className,
+      ...restHTMLProps
+    },
+    ref,
+  ) => {
+    const [isOpen, setIsOpen] = useState(defaultOpen);
+
+    const arrowRef = React.useRef<SVGSVGElement>(null);
+
+    const { refs, floatingStyles, context } = useFloating({
+      open: userOpen ?? isOpen,
+      onOpenChange: setIsOpen,
+      placement,
+      // Make sure the tooltip stays on the screen
+      whileElementsMounted: autoUpdate,
+      middleware: [
+        offset(ARROW_HEIGHT + GAP),
+        flip({
+          fallbackAxisSideDirection: 'start',
+        }),
+        shift(),
+        arrow({
+          element: arrowRef,
+        }),
+      ],
+    });
+
+    const { styles: animationStyles } = useTransitionStyles(context, {
+      initial: {
+        opacity: 0,
+      },
+    });
+
+    const { getReferenceProps, getFloatingProps } = useInteractions([
+      // Event listeners to change the open state
+      useHover(context, { move: false, delay }),
+      useFocus(context),
+      useDismiss(context),
+      // Role props for screen readers
+      useRole(context, { role: 'tooltip' }),
+    ]);
+
+    const mergedRef = useMemo(
+      () => mergeRefs([ref, refs.setFloating]),
+      [refs.setFloating, ref],
+    );
+
+    const childMergedRef = useMemo(
```

Comment:
Try using floating-ui's own [useMergeRefs](https://floating-ui.com/docs/react-utils#usemergerefs), its the same implementation anyway and follows the rules of [useMemo](https://react.dev/reference/react/useMemo).

We should stick to one way of merging refs and since we are already married to floating-ui and already using it other places 💒 🤷‍♂️